packages/next/src/server/lib/start-server.ts

```diff
@@ -362,9 +372,20 @@ export async function startServer(
 if (process.env.NEXT_PRIVATE_WORKER && process.send) {
   process.addListener('message', async (msg: any) => {
     if (msg && typeof msg && msg.nextWorkerOptions && process.send) {
-      await trace('start-dev-server').traceAsyncFn(() =>
+      startServerSpan = trace('start-dev-server', undefined, {
+        cpus: String(os.cpus().length),
+        platform: os.platform(),
+        'memory.freeMem': String(os.freemem()),
+        'memory.totalMem': String(os.totalmem()),
+        'memory.heapSizeLimit': String(v8.getHeapStatistics().heap_size_limit),
+      })
+      await startServerSpan.traceAsyncFn(() =>
         startServer(msg.nextWorkerOptions)
       )
+      const memoryUsage = process.memoryUsage()
+      startServerSpan.attrs['memory.rss'] = String(memoryUsage.rss)
```

Comment:
attrs is actually a private api, I think you can add a `setAttribute` method here: https://github.com/vercel/next.js/blob/ab457da3f148881825783cd77f2d21823a735890/packages/next/src/trace/trace.ts#L21

We should also add `protected` / `private` to those attributes so that they can't be used.

Reply:
Update: found we already have setAttribute https://github.com/vercel/next.js/blob/ab457da3f148881825783cd77f2d21823a735890/packages/next/src/trace/trace.ts#L103-L105